core/cas-server-core-webflow-mfa-api/src/main/java/org/apereo/cas/web/flow/actions/MfaInitializeAction.java

```diff
@@ -0,0 +1,41 @@
+package org.apereo.cas.web.flow.actions;
+
+import org.apereo.cas.services.MultifactorAuthenticationProvider;
+import org.apereo.cas.util.spring.ApplicationContextProvider;
+
+import lombok.val;
+
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * This Action is wired in by AbstractCasMultifactorWebflowConfigurer to put
+ * the resolved provider in the flow scope.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.4
+ */
+public class MfaInitializeAction extends AbstractAction {
+
+    @Override
+    protected Event doExecute(final RequestContext context) throws Exception {
+        val applicationContext = ApplicationContextProvider.getApplicationContext();
+        val activeFlow = context.getActiveFlow().getId();
+
+        val provider =
+                applicationContext
+                        .getBeansOfType(MultifactorAuthenticationProvider.class)
+                        .entrySet().stream().filter(e -> e.getKey().startsWith(activeFlow))
+                        .map(e -> e.getValue())
+                        .findFirst().get();
```

Comment:
Calling `get` on `Optional` without first checking with `isPresent` and co. might result in `NoSuchElementException`

Reply:
Thanks for the catch, pushed a change to correct.

Reply:
👍